src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultGeneratedDocumentPublisher.cs

```diff
@@ -61,18 +60,23 @@ public override void PublishCSharp(string filePath, SourceText sourceText, long
 
             _foregroundDispatcher.AssertForegroundThread();
 
-            if (!_publishedCSharpSourceText.TryGetValue(filePath, out var previouslyPublishedText))
+            if (!_publishedCSharpData.TryGetValue(filePath, out var previouslyPublishedData))
             {
-                previouslyPublishedText = EmptySourceText;
+                previouslyPublishedData = PublishData.Default;
             }
 
             IReadOnlyList<TextChange> textChanges = Array.Empty<TextChange>();
-            if (!sourceText.ContentEquals(previouslyPublishedText))
+            if (!sourceText.ContentEquals(previouslyPublishedData.SourceText))
```

Comment:
Could just make `PublishData` IEquatable  to simplify this.

Reply:
Thought about that but it'd have to be IEquatable with SourceText or we'd have to create a fake PublishData for comparison purposes.